### PR TITLE
Better wording for messages around disabled optimizations

### PR DESCRIPTION
**Pull request by lptr**
Supersedes https://github.com/gradle/gradle/pull/15990.

### CODE REVIEW

**Comment by lptr** [comment]
@bot-gradle test QF please.

**Comment by ldaley** [comment]
https://github.com/gradle/gradle/blob/f59df7d2a01ae6971cb37c13cd60dcb17e24afe7/subprojects/execution/src/main/java/org/gradle/internal/execution/steps/ResolveChangesStep.java#L43 is out of alignment after this change. This shows up in info logging and build scans.

**Comment by ldaley** [comment]
https://github.com/gradle/gradle/blob/f59df7d2a01ae6971cb37c13cd60dcb17e24afe7/subprojects/execution/src/main/java/org/gradle/internal/execution/steps/ResolveCachingStateStep.java#L53

is out of alignment after this change. This shows up in info logging.

**Review by wolfs — APPROVED** [review]
LGTM!

**Comment by lptr** [comment]
@bot-gradle test QFL please.

**Comment by lptr** [comment]
@bot-gradle test QFL please.

File: subprojects/core/src/integTest/groovy/org/gradle/api/internal/tasks/TaskCacheabilityReasonIntegrationTest.groovy [hunk 1]
```diff
@@ -339,7 +339,7 @@ class TaskCacheabilityReasonIntegrationTest extends AbstractIntegrationSpec impl
 
         then:
         withBuildCache().succeeds("producer", "consumer")
-        assertCachingDisabledFor VALIDATION_FAILURE, "Validation failed", ":consumer"
+        assertCachingDisabledFor VALIDATION_FAILURE, "Caching has been disabled to ensure correctness due to validation failures. Please consult deprecation warnings for more details.", ":consumer"
```
Comment (wolfs) [inline subprojects/core/src/integTest/groovy/org/gradle/api/internal/tasks/TaskCacheabilityReasonIntegrationTest.groovy:342]:
```suggestion
        assertCachingDisabledFor VALIDATION_FAILURE, "Caching has been disabled due to validation failures to ensure correctness. Please consult deprecation warnings for more details.", ":consumer"
```

It sounds much better to me if the `due to` comes first.

**Review by wolfs — COMMENTED** [review]
I left a comment.

**Comment by lptr** [comment]
@bot-gradle test QFL please.

File: subprojects/core/src/integTest/groovy/org/gradle/api/internal/tasks/TaskCacheabilityReasonIntegrationTest.groovy (hunk 1 above)
Comment (lptr) [reply subprojects/core/src/integTest/groovy/org/gradle/api/internal/tasks/TaskCacheabilityReasonIntegrationTest.groovy]: I agree, changed it so.

**Comment by lptr** [comment]
@bot-gradle test QFL please.

File: subprojects/execution/src/test/groovy/org/gradle/internal/execution/steps/ResolveCachingStateStepTest.groovy [hunk 2]
```diff
@@ -52,7 +52,7 @@ class ResolveCachingStateStepTest extends StepSpec<BeforeExecutionContext> {
         _ * context.validationProblems >> Optional.of({ ImmutableList.of("Validation problem") } as ValidationContext)
         1 * delegate.execute(work, { CachingContext context ->
             context.cachingState.disabledReasons*.category == [CachingDisabledReasonCategory.VALIDATION_FAILURE]
-            context.cachingState.disabledReasons*.message == ["Validation failed"]
+            context.cachingState.disabledReasons*.message == ["Caching has been disabled due to validation failures to ensure correctness. Please consult deprecation warnings for more details."]
```
Comment (ldaley) [inline subprojects/execution/src/test/groovy/org/gradle/internal/execution/steps/ResolveCachingStateStepTest.groovy:55]: Suggestion:

```
Caching has been disabled to ensure correctness. Please consult deprecation warnings for more details.
```

As discussed, the term/concept of “validation failures” isn't meaningful to users an potentially confusing. I don't think it adds anything here and can be safely removed.

File: subprojects/execution/src/main/java/org/gradle/internal/execution/steps/ResolveChangesStep.java [hunk 3]
```diff
@@ -40,7 +40,7 @@
 
 public class ResolveChangesStep<R extends Result> implements Step<CachingContext, R> {
     private static final String NO_HISTORY = "No history is available.";
-    private static final String VALIDATION_FAILED = "Validation failed.";
+    private static final String VALIDATION_FAILED = "Change tracking has been disabled due to validation failures to ensure correctness. Please consult deprecation warnings for more details.";
```
Comment (ldaley) [inline subprojects/execution/src/main/java/org/gradle/internal/execution/steps/ResolveChangesStep.java:43]: Suggestion:

```
Change tracking has been disabled to ensure correctness. Please consult deprecation warnings for more details.
```

As discussed, the term/concept of “validation failures” isn't meaningful to users an potentially confusing. I don't think it adds anything here and can be safely removed.

Comment (ldaley) [inline subprojects/execution/src/main/java/org/gradle/internal/execution/steps/ResolveChangesStep.java:43]: Suggestion:

```
Change tracking has been disabled to ensure correctness. Please consult deprecation warnings for more details.
```

As discussed, the term/concept of “validation failures” isn't meaningful to users an potentially confusing. I don't think it adds anything here and can be safely removed.

File: subprojects/execution/src/main/java/org/gradle/internal/execution/steps/ResolveCachingStateStep.java [hunk 4]
```diff
@@ -50,7 +50,7 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(ResolveCachingStateStep.class);
     private static final CachingDisabledReason BUILD_CACHE_DISABLED_REASON = new CachingDisabledReason(CachingDisabledReasonCategory.BUILD_CACHE_DISABLED, "Build cache is disabled");
     private static final CachingState BUILD_CACHE_DISABLED_STATE = CachingState.disabledWithoutInputs(BUILD_CACHE_DISABLED_REASON);
-    private static final CachingDisabledReason VALIDATION_FAILED_REASON = new CachingDisabledReason(CachingDisabledReasonCategory.VALIDATION_FAILURE, "Validation failed");
+    private static final CachingDisabledReason VALIDATION_FAILED_REASON = new CachingDisabledReason(CachingDisabledReasonCategory.VALIDATION_FAILURE, "Caching has been disabled due to validation failures to ensure correctness. Please consult deprecation warnings for more details.");
```
Comment (ldaley) [inline subprojects/execution/src/main/java/org/gradle/internal/execution/steps/ResolveCachingStateStep.java:53]: Suggestion:

Caching has been disabled to ensure correctness. Please consult deprecation warnings for more details.
As discussed, the term/concept of “validation failures” isn't meaningful to users an potentially confusing. I don't think it adds anything here and can be safely removed.

**Review by ldaley — CHANGES_REQUESTED** [review]

**Comment by lptr** [comment]
@bot-gradle test QFL please.

**Review by ldaley — APPROVED** [review]

**Review by melix — APPROVED** [review]

**Pull request merged**